pkg/registry/registry.go

```diff
@@ -475,34 +482,86 @@ func (r *PlainRegistry) listArchsWithAuth(ctx context.Context, client http.Clien
 	platforms := []Platform{}
 	switch resp.Header.Get("Content-Type") {
 	case "application/vnd.docker.distribution.manifest.list.v2+json", "application/vnd.oci.image.index.v1+json":
+		outChan := make(chan struct {
+			Platform Platform
+			err      error
+		})
```

Comment:
This should probably declared as an internal type so we avoid the structure definition repetition all around the code.